### PR TITLE
isso: 0.12.6.2 -> 0.13.0

### PR DESCRIPTION
**Pull request by RaitoBezarius**
###### Description of changes

This uses `buildNpmPackage` now and Node.js 20.
Also, packages the HTML documentation in a new output.

https://isso-comments.de/news/#isso-version-0-13-0-released

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by RaitoBezarius** [comment]
Contributes towards #229910.

File: pkgs/servers/isso/default.nix [hunk 1]
```diff
@@ -1,30 +1,40 @@
...
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out/
+      cp -r node_modules $out/node_modules
+
+      runHook postInstall
+    '';
+  };
```
Comment (JulienMalka) [inline pkgs/servers/isso/default.nix:31]:
```suggestion
  npmDeps = fetchNpmDeps {
    inherit src;
    hash = "sha256-RBpuhFI0hdi8bB48Pks9Ac/UdcQ/DJw+WFnNj5f7IYE=";
  };
```

File: pkgs/servers/isso/default.nix [hunk 2]
```diff
@@ -33,16 +43,19 @@ with python3Packages; buildPythonApplication rec {
...
+    sphinx
   ];
 
   nativeBuildInputs = [
     cffi
+    sphinxHook
     nodejs
   ];
 
+  NODE_PATH = "$nodeDependencies";
```
Comment (JulienMalka) [inline pkgs/servers/isso/default.nix:55]:
```suggestion
  NODE_PATH = "$npmDeps";
```

File: pkgs/servers/isso/default.nix [hunk 3]
```diff
@@ -1,30 +1,40 @@
...
-  };
-  nodePackages = import ./node-packages.nix {
-    inherit (pkgs) fetchurl nix-gitignore stdenv lib fetchgit;
-    inherit nodeEnv;
-  };
-
-  nodeDependencies = (nodePackages.shell.override (old: {
-  })).nodeDependencies;
-in
+{ pkgs, nodejs, buildNpmPackage, lib, python3Packages, fetchFromGitHub, nixosTests }:
```
Comment (JulienMalka) [inline pkgs/servers/isso/default.nix:1]:
```suggestion
{ pkgs, nodejs, lib, python3Packages, fetchFromGitHub, nixosTests, fetchNpmDeps, npmHooks }:
```

File: pkgs/servers/isso/default.nix [hunk 4]
```diff
@@ -33,16 +43,19 @@ with python3Packages; buildPythonApplication rec {
...
+    sphinxHook
     nodejs
   ];
 
+  NODE_PATH = "$nodeDependencies";
+
   preBuild = ''
-    ln -s ${nodeDependencies}/lib/node_modules ./node_modules
-    export PATH="${nodeDependencies}/bin:$PATH"
+    export PATH="$nodeDependencies/node_modules/.bin:$PATH"
```
Comment (JulienMalka) [inline pkgs/servers/isso/default.nix:58]:
```suggestion
    ln -s ${npmDeps}/node_modules ./node_modules
    export PATH="${npmDeps}/bin:$PATH"
```

**Review by JulienMalka — CHANGES_REQUESTED** [review]
Looks like using `fetchNpmDeps` is what you are looking for here :)

**Comment by JulienMalka** [comment]
Result of `nixpkgs-review pr 233399` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>isso</li>
    <li>isso.dist</li>
    <li>isso.doc</li>
  </ul>
</details>

**Review by JulienMalka — APPROVED** [review]
Looks good to me 🥖

File: pkgs/servers/isso/default.nix [hunk 5]
```diff
@@ -33,16 +29,21 @@ with python3Packages; buildPythonApplication rec {
     werkzeug
     bleach
     flask-caching
+    sphinx
```
Comment (SuperSandro2000) [inline pkgs/servers/isso/default.nix:32]: If this is used by the sphinxHook, it goes into nativeBuildInputs, too

File: pkgs/servers/isso/default.nix [hunk 6]
```diff
@@ -1,30 +1,26 @@
...
-    inherit nodeEnv;
-  };
-
-  nodeDependencies = (nodePackages.shell.override (old: {
-  })).nodeDependencies;
-in
+{ pkgs, nodejs, lib, python3Packages, fetchFromGitHub, nixosTests, fetchNpmDeps, npmHooks }:
 with python3Packages; buildPythonApplication rec {
 
   pname = "isso";
```
Comment (SuperSandro2000) [inline pkgs/servers/isso/default.nix:4]:
```suggestion
{ pkgs, nodejs, lib, python3Packages, fetchFromGitHub, nixosTests, fetchNpmDeps, npmHooks }:

with python3Packages;

buildPythonApplication rec {
  pname = "isso";
```
nit

File: pkgs/servers/isso/default.nix (hunk 5 above)
Comment (RaitoBezarius) [reply pkgs/servers/isso/default.nix:32]: right

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 233399` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>isso</li>
    <li>isso.dist</li>
    <li>isso.doc</li>
  </ul>
</details>

**Pull request merged**